src/arena.c

```diff
@@ -1366,7 +1366,9 @@ arena_malloc_hard(tsdn_t *tsdn, arena_t *arena, size_t size, szind_t ind,
 		assert(sz_can_use_slab(size));
 		return arena_malloc_small(tsdn, arena, ind, zero);
 	} else {
-		return large_malloc(tsdn, arena, sz_index2size(ind), zero);
+		size_t usize = sz_limit_usize_gap_enabled()? sz_s2u(size):
+		    sz_index2size(ind);
```

Comment:
Always do s2u, since that will be the long term solution anyway?

Reply:
I am also thinking about this. If we always use `sz_s2u`, there will be some redundant calculation since we already calculated `ind` before. I am not sure compiler can optimize this for us. WDYT?

Reply:
This is the large path, so using the index2size is only possible in the short term. Large s2u is very efficient anyway (simply round up to the next page?)